modules/helpers/dice-helpers.js

```diff
@@ -57,12 +57,20 @@ export default class DiceHelpers {
     // Determine if this roll is triggered by an item.
     let item = {};
     if ($(row.parentElement).hasClass("item")) {
-      let itemID = row.parentElement.dataset["itemId"];
-      const item1 = actor.getOwnedItem(itemID);
-      item = Object.entries(data.items).filter((item) => item[1]._id === itemID);
-      item = item[0][1];
-      item.flags.uuid = item1.uuid;
-    }
+      //Check if token is linked to actor
+      if (obj.actor.token === null) {
```

Comment:
This should work, but if we're dealing with a Foundry Actor entity we can use the helper function isToken() as best-practice. Maybe worth a follow-up.